e2e/authorization_test.go

```diff
@@ -0,0 +1,164 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/GoogleCloudPlatform/prometheus-engine/e2e/kubeutil"
+	"github.com/GoogleCloudPlatform/prometheus-engine/e2e/operatorutil"
+	"github.com/GoogleCloudPlatform/prometheus-engine/pkg/operator"
+	monitoringv1 "github.com/GoogleCloudPlatform/prometheus-engine/pkg/operator/apis/monitoring/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func TestTLS(t *testing.T) {
+	t.Parallel()
+	tctx := newOperatorContext(t)
+	ctx := context.Background()
+
+	tctx.createOperatorConfigFrom(ctx, monitoringv1.OperatorConfig{
+		Features: monitoringv1.OperatorFeatures{
+			TargetStatus: monitoringv1.TargetStatusSpec{
+				Enabled: true,
+			},
+		},
+	})
+
+	c := tctx.Client()
+	const appName = "tls-insecure"
+	deployment, err := operatorutil.SyntheticAppDeploy(ctx, c, tctx.namespace, appName, []string{
+		"--tls-create-self-signed=true",
+	})
+	if err != nil {
+		tctx.Fatal(err)
+	}
+	if err := kubeutil.WaitForDeploymentReady(ctx, c, tctx.namespace, appName); err != nil {
+		kubeutil.DeploymentDebug(tctx.T, ctx, tctx.RestConfig(), tctx.Client(), tctx.namespace, appName)
+		tctx.Fatalf("failed to start app: %s", err)
+	}
+
+	tctx.Run("no-tls", tctx.subtest(func(ctx context.Context, t *OperatorContext) {
```

Comment:
```suggestion
	tctx.Run("untrusted-tls-ca", tctx.subtest(func(ctx context.Context, t *OperatorContext) {
```
To be a bit more specific?

Reply:
or `tls-failed`